### PR TITLE
FIX: honor ``TAG_REMOTE`` from the previous ``TAG`` activity

### LINKED ISSUE #108: BUG: updated github authentication was ignored by program
behavior was as follows:
1) tried to run rever for a new release
2) fixed multiple errors in various places and kept rerunning....
3) finally got the github push which failed
4) realized that in the config the push authentication was given as ssh but my account is set up using https
5) changed the authentication command in setup.py to the https command
6) retried rever but it kept looking for the ssh authentication for some reason

@CJ-Wright is looking into this, but I am reporting for completeness

### PR DESCRIPTION
**Pull request by CJ-Wright**
with a warning
Closes #108

### CODE REVIEW

**Comment by CJ-Wright** [comment]
ATTN: @sbillinge

**Comment by CJ-Wright** [comment]
@scopatz ready for review.

File: rever/activities/push_tag.xsh [hunk 1]
```diff
@@ -31,6 +32,12 @@ class PushTag(Activity):
                          desc="Tags the current version.")
 
     def _func(self, remote=None, target=None):
+        if remote is None:
+            remote = ${...}.get('TAG_REMOTE', None)
+            warnings.warn(DeprecationWarning('TAG_REMOTE has been deprecated '
```
Comment (scopatz) [inline rever/activities/push_tag.xsh]: This warning is issued even if TAG_REMOTE doesn't exist.

**Comment by scopatz** [comment]
HI @CJ-Wright I am a little torn about this. I think that it is easier and better to just fix the existed rever installs out there than to add a deprecation warning and encourage people to not change it immediately.

**Comment by scopatz** [comment]
Thanks for putting this in!

File: rever/activities/push_tag.xsh (hunk 1 above)
Comment (CJ-Wright) [reply rever/activities/push_tag.xsh]: Yes, that was silly sorry.

**Comment by CJ-Wright** [comment]
My issue is tat it fails silently. Rever will try to use the org/repo settings without letting the user know, which lead to some confusion on my side.

**Comment by scopatz** [comment]
Sure, so maybe we should raise a RuntimeError in the case were TAG_REMOTE is still defined with a message that says this has been deprecated, please change to using PUSH_TAG_REMOTE

**Comment by CJ-Wright** [comment]
I can work with that.

**Comment by CJ-Wright** [comment]
@scopatz ready for review!

File: rever/activities/push_tag.xsh [hunk 2]
```diff
@@ -31,6 +32,11 @@ class PushTag(Activity):
                          desc="Tags the current version.")
 
     def _func(self, remote=None, target=None):
+        if 'TAG_REMOTE' in ${...}:
+            warnings.warn(DeprecationWarning('TAG_REMOTE has been deprecated '
```
Comment (scopatz) [inline rever/activities/push_tag.xsh]: I don't think that this follows the API of warn(). https://docs.python.org/3.6/library/warnings.html#warnings.warn

**Comment by CJ-Wright** [comment]
I think #154 is a better solution at this point.

**Pull request closed without merging**